src/main/fox_demo.c

```diff
@@ -0,0 +1,3005 @@
+extern void func_80197290(Player*);
+extern void func_8019C85C(Player*);
+extern void func_8019D76C(Player*);
+extern void func_8019EE60(Player*);
+extern void func_801A7930(Player*);
+extern void func_801AF8F4(Player*, OSContPad**, OSContPad**);
+extern void func_801B3D04(Player*, OSContPad**, OSContPad**);
+extern void func_8002ED60(Player*);
+extern void func_800935E8(Player*);
+extern void func_8018769C(Player*);
+extern void func_8018CD8C(Player*);
+extern void func_8018DF74(Player*);
+extern void func_8018EF6C(Player*);
+extern void func_8018F880(Player*);
+extern void func_80192E20(Player*);
+extern void func_80193A30(Player*);
+extern void func_80196BF8(Player*);
+extern void func_801988B8(Player*);
+extern void func_8019BA64(Player*);
+extern void func_8019D428(Player*);
+extern void func_801A0AC0(Player*);
+extern void func_801A10F4(Player*);
+extern void func_801AB9B0(Player*);
+extern void func_801AC754(Player*);
+extern void func_800A46A0(Player*);
+extern void func_800AD7F0(Player*);
+extern void func_800B2574(Player*);
+extern void func_800AE278(Player*);
+extern void func_80093164(Object_2F4*);
+extern void func_800A6028(Vec3f*, u32);
+extern void func_8018DA58(Object_2F4*);
+extern void func_8018ED9C(Object_2F4*);
+extern void func_80195E44(Object_2F4*);
+extern void func_80197F10(Object_2F4*);
+extern void func_8019DD20(Object_2F4*);
+extern void func_8019E5F0(Object_2F4*);
+extern void func_8019FF00(Object_2F4*);
+extern void func_801A8BE8(Object_2F4*);
+extern void func_801B28BC(Object_2F4*);
+
+void func_80048AC0(s32 arg0) {
+    s32 teamShield;
+
+    if (arg0 == 1) {
+        if (gTeamShields[1] < 0x40) {
+            func_800BA808(gMsg_ID_20303, 0xA);
+            return;
+        }
+        if ((s32) (Play_GetMaxShields() * 0.8f) < gPlayer->shields) {
+            func_800BA808(gMsg_ID_20017, 0xA);
+            return;
+        }
+        if ((s32) (Play_GetMaxShields() * 0.3f) < gPlayer->shields) {
+            func_800BA808(gMsg_ID_20019, 0xA);
+            return;
+        }
+        func_800BA808(gMsg_ID_20018, 0xA);
+        return;
+    }
+
+    teamShield = gTeamShields[arg0];
+
+    if (teamShield >= 0xA1) {
+        switch (arg0) {
+            case 2:
+                func_800BA808(gMsg_ID_20011, 0x14);
+                return;
+            case 3:
+                func_800BA808(gMsg_ID_20012, 0x1E);
+                return;
+        }
+    } else if (teamShield >= 0x41) {
+        switch (arg0) {
+            case 2:
+                func_800BA808(gMsg_ID_20013, 0x14);
+                return;
+            case 3:
+                func_800BA808(gMsg_ID_20014, 0x1E);
+                return;
+        }
+    } else {
+        switch (arg0) {
+            case 2:
+                func_800BA808(gMsg_ID_20015, 0x14);
+                return;
+            case 3:
+                func_800BA808(gMsg_ID_20016, 0x1E);
+                return;
+        }
+    }
+}
+
+void func_80048CC4(Object_2F4* arg0, s32 arg1) {
```

Comment:
arg0 should be named obj2F4 to make it easier to change struct fields later. Similarly for other arguments of object type.